File: Python/2130.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def pairSum(self, head):
        lst = []
        while(head):
            lst.append(head.val)
            head = head.next
        max_ = -float("inf")
        n = len(lst)
        for i in range(n//2):
            if lst[i] + lst[n-i-1] > max_:
                max_ = lst[i] + lst[n-i-1]
        return max_

File: Java/2130.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
        int max = Integer.MIN_VALUE;
        List<Integer> lst = new ArrayList<>();
        while(head != null){
            lst.add(head.val);
            head = head.next;
        }
        int n = lst.size();
        for(int i = 0; i < n/2; i++){
            if(lst.get(i) + lst.get(n-i-1) > max) max = lst.get(i) + lst.get(n-i-1);
        }
        return max;
    }
}